apps/sms/views/shared/js/utils.js

```diff
@@ -473,6 +473,25 @@
       });
       return parsed;
     },
+    url(base, params) {
+      if (base.indexOf('?') === -1) {
+        base += '?';
+      } else {
+        base += '&';
+      }
+
+      for (var key in params) {
+        if (params[key] == null) { // null or undefined
+          continue;
+        }
+
+        base +=
+          encodeURIComponent(key) + '=' +
+          encodeURIComponent(params[key]) + '&';
```

Comment:
Here is a proposal:
* when starting an activity (or a new message from body -- and likely forward, I haven't tried), store the activity data inside a service
* the service returns an id
* we redirect to the new page using this id
* we recover the data using the id -- and the data is deleted from the service

How does it sound ? Is it over-engineered ?